packages/ui/src/ui/containers/MaintenancePage/HandleMaintenance.tsx

```diff
@@ -1,62 +1,51 @@
-import {withRouter} from 'react-router-dom';
-import React, {Component} from 'react';
-import {Redirect} from 'react-router';
-import {connect} from 'react-redux';
-import PropTypes from 'prop-types';
-import {compose} from 'redux';
-
-import MaintenancePage from '../../containers/MaintenancePage/MaintenancePage';
-
-import {Page} from '../../constants/index';
-
-class HandleRedirect extends Component {
-    static propTypes = {
-        location: PropTypes.object.isRequired,
-        cluster: PropTypes.string,
-        eventsFirstUpdate: PropTypes.bool.isRequired,
-        maintenancePageEvent: PropTypes.object,
-    };
-
-    shouldComponentUpdate(nextProps) {
-        const {location} = this.props;
-        const {eventsFirstUpdate, location: nextLocation} = nextProps;
-        const isDifferentLocation = location !== nextLocation;
-
-        return isDifferentLocation || eventsFirstUpdate;
-    }
+import React from 'react';
+import {ConnectedProps, connect} from 'react-redux';
+
+import {MaintenancePage} from '../../containers/MaintenancePage/MaintenancePage';
+import {RootState} from '../../store/reducers';
+import {isMaintenanceIgnored, setMaintenanceIgnored} from '../../utils/maintenance';
+
+type Props = {
+    cluster: string;
+    children: React.ReactNode;
+    maintenanceContent?: React.ReactNode;
+};
 
+type ReduxProps = ConnectedProps<typeof connector>;
+
+class HandleMaintenance extends React.Component<Props & ReduxProps> {
     render() {
-        const {location, cluster, maintenancePageEvent} = this.props;
+        const {cluster, maintenancePageEvent, maintenanceContent, children} = this.props;
 
```

Comment:
I would do this `if (! maintenancePageEvent || isIgnored) return children; ` this would save the code from 1 level of nesting